testplan/runnable/base.py

```diff
@@ -733,7 +796,7 @@ def schedule_all(
 
         tasks = self.discover(path=path, name_pattern=name_pattern)
         for task in tasks:
-            self.add(task, resource=resource)
+            self._add(task, resource=resource)
```

Comment:
The only thing I don't particularly like is here, that we can no longer pipe discover & add which are the 2 major public APIs.